src/Tools/dotnet-trace/ProviderUtils.cs

```diff
@@ -55,9 +58,50 @@ internal static class Extensions
             { "compilationdiagnostic", 0x2000000000 },
             { "methoddiagnostic", 0x4000000000 },
             { "typediagnostic", 0x8000000000 },
+            { "jitinstrumentationdata", 0x10000000000 },
+            { "profiler", 0x20000000000 },
             { "waithandle", 0x40000000000 },
+            { "allocationsampling", 0x80000000000 },
         };
 
+        private enum ProviderSource
+        {
+            ProvidersArg = 1,
+            CLREventsArg = 2,
+            ProfileArg = 4,
+        }
+
+        public static void MergeProfileAndProviders(Profile selectedProfile, List<EventPipeProvider> providerCollection, Dictionary<string, string> enabledBy)
```

Comment:
Is this dead code? I saw you removed the call in collect

Reply:
Yes, I just needed to remove instances from the tests, which I did in addition to revamping the tests.